File: src/main/java/sage/transfer/UserCardFollow.java
package sage.transfer;

import java.util.Collection;

import sage.entity.Follow;
import sage.entity.Tag;
import sage.util.Colls;

public class UserCardFollow {
  public String reason;
  public Collection<Long> tagIds;
  public boolean includeNew;
  public boolean includeAll;

  UserCardFollow() {}

  public UserCardFollow(Follow follow) {
    reason = follow.getReason();
    tagIds = Colls.map(follow.getTags(), Tag::getId);
    includeNew = follow.isIncludeNew();
    includeAll = follow.isIncludeAll();
  }
}

File: src/main/kotlin/sage/transfer/UserCardFollow.kt
package sage.transfer

import sage.annotation.KotlinNoArg
import sage.entity.Follow

@KotlinNoArg
class UserCardFollow {
  var reason: String? = null
  var tagIds: Collection<Long> = emptyList()
  var includeNew: Boolean = false
  var includeAll: Boolean = false

  constructor(follow: Follow) {
    reason = follow.reason
    tagIds = follow.tags.map { it.id }
    includeNew = follow.isIncludeNew
    includeAll = follow.isIncludeAll
  }
}
